### PR TITLE
CDPSDX-3528; secure the communication on the hdfs. Therefore I added…

### PR DESCRIPTION
**Pull request by RokolyA94**
secure the communication on the `hdfs`. Therefore I added new values to the Role config. The new values are the `rpc protection`, and the `encrypt data transfer`.

Tested with `cbd` the new values set correctly.

### CODE REVIEW

**Review by kkalvagadda1 — COMMENTED** [review]
Looks good to me.

**Review by yg-gd — APPROVED** [review]

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java [hunk 1]
```diff
@@ -49,6 +51,11 @@ protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, Templat
                             config(DFS_REPLICATION, DFS_REPLICATION_VALUE.toString())
                     );
                 }
+                if (entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId())) {
```
Comment (lnardai) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java:54]: The previous block is also about `isSDXOptimizationEnabled`. This block check that entitlement but is also applied for Datahubs, this is confusing. I would clean up the previous method name seems that looks more specific.

**Review by lnardai — CHANGES_REQUESTED** [review]
Clean up logic for isSDXOptimizedConfigurationEnabled vs. entitlement check in CoreConfig.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java (hunk 1 above)
Comment (RokolyA94) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java:54]: https://jira.cloudera.com/browse/CDPSDX-3528?focusedCommentId=5522462&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-5522462

**Comment by RokolyA94** [comment]
@lnardai 
based on the `Chirs Royles` comment on the JIRA ticket. These changes should be applied to `datahubs` and `datalakes`.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java (hunk 1 above)
Comment (kkalvagadda1) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java:54]: @RokolyA94 This change is limited to only SDX. Please update the code to add the check.

**Review by kkalvagadda1 — CHANGES_REQUESTED** [review]

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java (hunk 1 above)
Comment (RokolyA94) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java]: stack type check added.
restricted only for DL.

**Review by romedeiro — APPROVED** [review]

**Review by kkalvagadda1 — APPROVED** [review]
Looks good.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/core/CoreConfigProvider.java [hunk 2]
```diff
@@ -46,8 +50,15 @@ public class CoreConfigProvider extends AbstractRoleConfigProvider {
     @Inject
     private S3ConfigProvider s3ConfigProvider;
 
+    @Inject
+    private EntitlementService entitlementService;
+
     @Override
     protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
+        if (entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId())
```
Comment (doktoric) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/core/CoreConfigProvider.java:58]: Please add some unit test coverage for this.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java [hunk 3]
```diff
@@ -44,11 +46,16 @@ protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, Templat
...
                 );
-                if (isSDXOptimizationEnabled(source)) {
+                if (isSDXOptimizationEnabled(source) && isNamenodeHA(source)) {
                     configs.add(
                             config(DFS_REPLICATION, DFS_REPLICATION_VALUE.toString())
                     );
                 }
+                if (isSDXOptimizationEnabled(source)) {
+                    configs.add(
+                            config(DFS_ENCRYPT_DATA_TRANSFER, "true")
```
Comment (doktoric) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java:56]: where is the unit test for this.

**Review by doktoric — APPROVED** [review]

**Comment by lnardai** [comment]
Could you fix up the commit message, looks like some typo after the ticket id.
CB-xxxxx relevant text here
or
CDPSDX-xxxxxx relevant text here

**Comment by RokolyA94** [comment]
@lnardai 
commit message fixed.

**Review by doktoric — APPROVED** [review]

**Pull request merged**